### PR TITLE
Change mock based on connection test with HMCTS

### PR DESCRIPTION
**Pull request by archferns**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/CACP-79)

Describe what you did and why.
Changed the mock setup to match the URLs, secrets, and methods used by HMCTS in the connection test. This will allow us to run the mock as a backend to CDA in its current state.

## Checklist

Before you ask people to review this PR:

- [x] Tests and linters should be passing
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/controllers/laa_references_controller_spec.rb [hunk 1]
```diff
@@ -73,19 +74,24 @@
...
     end
 
+    it 'returns unauthorized' do
+      put :record_representation_order, params: laa_reference_params
+      expect(response).to be_unauthorized
+    end
+
     context 'when the LaaReference exists' do
-      let(:laa_reference_id) { laa_reference.id }
+      before { laa_reference }
```
Comment (mpw5) [inline spec/controllers/laa_references_controller_spec.rb]: Should this be `laa_reference.save!` as it is in the other specs?

Comment (archferns) [reply spec/controllers/laa_references_controller_spec.rb]: Yep, that's better, good catch :+1:

File: spec/controllers/laa_references_controller_spec.rb [hunk 2]
```diff
@@ -28,39 +31,37 @@
...
     end
 
+    it 'returns unauthorized' do
+      put :record_reference, params: laa_reference_params
+      expect(response).to be_unauthorized
+    end
+
     context 'when the LaaReference exists' do
-      let(:laa_reference_id) { laa_reference.id }
+      before { laa_reference }
```
Comment (archferns) [inline spec/controllers/laa_references_controller_spec.rb]:
```suggestion
      before { laa_reference.save! }
```

File: spec/controllers/laa_references_controller_spec.rb (hunk 1 above)
Comment (archferns) [inline spec/controllers/laa_references_controller_spec.rb]:
```suggestion
      before { laa_reference.save! }
```

**Pull request merged**